flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -217,12 +218,27 @@ private static HiveConf createHiveConf(
                     HadoopUtils.possibleHadoopConfPaths(
                             new org.apache.flink.configuration.Configuration())) {
                 hadoopConf = getHadoopConfiguration(possibleHadoopConfPath);
-                if (hadoopConf != null) {
+                if (hadoopConf != null && hadoopConf.size() != 0) {
```

Comment:
Why do we check the size of the conf? Note that `size()` is the number of keys in the configuration, not the added resources.

Reply:
if possiable path is exist without no conf files ,  hadoopConf  is not null  but  size is 0.
We can find other possiable path.

Reply:
I think the size is 0 when none of conf files exist, which should be correct.

Reply:
Hi @lirui-apache , If you feel something is wrong, I am trying to correct it.

Reply:
> I think the size is 0 when none of conf files exist, which should be correct.

No, we don't get such guarantees from the `size()` API contract so it's not safe to rely on it. We can modify `getHadoopConfiguration` to explicitly return null when the conf files don't exist.